corehq/blobs/tests/test_export.py

```diff
@@ -1,123 +1,90 @@
+            self.exporter.migrate(out.name, force=True)
+            with tarfile.open(out.name, 'r:gz') as tgzfile:
+                self.assertCountEqual({expected_meta.key}, tgzfile.getnames())
 
-        for type_code in [CODES.form_xml, CODES.multimedia, CODES.data_export]:
-            for domain in (cls.domain_name, str(uuid.uuid4())):
-                meta = cls.db.put(BytesIO(data), meta=new_meta(domain=domain, type_code=type_code))
-                lost = new_meta(domain=domain, type_code=type_code, content_length=42)
-                cls.blob_metas.append(meta)
-                cls.blob_metas.append(lost)
-                lost.save()
-                cls.not_found.add(lost.key)
+    def test_different_blob_types_are_exported(self):
+        form = self.db.put(BytesIO(self.blob_data), meta=new_meta(domain=self.domain, type_code=CODES.form_xml))
+        multimedia = self.db.put(
+            BytesIO(self.blob_data), meta=new_meta(domain=self.domain, type_code=CODES.multimedia)
+        )
 
-    @classmethod
-    def tearDownClass(cls):
-        for blob in cls.blob_metas:
-            blob.delete()
-        cls.db.close()
-        super().tearDownClass()
-
-    def test_migrate_all(self):
-        expected = {
-            m.key for m in self.blob_metas
-            if m.domain == self.domain_name and m.key not in self.not_found
-        }
         with NamedTemporaryFile() as out:
-            exporter = EXPORTERS['all_blobs'](self.domain_name)
-            exporter.migrate(out.name, force=True)
+            self.exporter.migrate(out.name, force=True)
             with tarfile.open(out.name, 'r:gz') as tgzfile:
-                self.assertEqual(expected, set(tgzfile.getnames()))
-
-    def test_migrate_multimedia(self):
-        image_path = os.path.join('corehq', 'apps', 'hqwebapp', 'static', 'hqwebapp', 'images',
-                                  'commcare-hq-logo.png')
-        with open(image_path, 'rb') as f:
-            image_data = f.read()
-
-        files = (
-            (CommCareImage, self.domain_name, image_data),
-            (CommCareAudio, self.domain_name, b'fake audio'),
-            (CommCareVideo, self.domain_name, b'fake video'),
-            (CommCareAudio, 'other_domain', b'fake audio 1'),
-        )
+                self.assertCountEqual({form.key, multimedia.key}, tgzfile.getnames())
 
-        blob_keys = []
-        for doc_class, domain, data in files:
-            obj = doc_class.get_by_data(data)
-            obj.attach_data(data)
-            obj.add_domain(domain)
-            self.addCleanup(obj.delete)
-            self.assertEqual(data, obj.get_display_file(False))
-            blob_keys.append(obj.blobs[obj.attachment_id].key)
+    def test_orphaned_blob_meta_is_not_exported(self):
+        expected_meta = self.db.put(
+            BytesIO(self.blob_data), meta=new_meta(domain=self.domain, type_code=CODES.form_xml)
+        )
+        self.addCleanup(self.db.delete, expected_meta.key)
```

Comment:
Why is `addCleanup` necessary here? Will that not be done automatically on fixture teardown?

Reply:
Addressed in b163a09933a06bf665cb817ed2b67e5300e02699